WordPress/gutenberg
# Blocks: Update default block categories

Closes #11406

This pull request seeks to propose to replace the existing set of default block categories with a new set of categories as proposed in #11406.

The updated set of block categories are:

- Text
- Media
- Design
- ~Tools~ **Edit:** Widgets _(Unchanged)_
- Embeds _(Unchanged)_
- Reusable _(Unchanged)_

![image](https://user-images.githubusercontent.com/1779930/71267560-e23c2080-2318-11ea-8eee-c56e8eb26db4.png)

<p><strong id="implementation-notes">Implementation Notes:</strong></p>

There is an attempt to preserve some backwards-compatibility here, based on the summarized findings at https://github.com/WordPress/gutenberg/issues/11406#issuecomment-567164861 .

Blocks which register to one of the existing categories will transparently map to one of the updated categories. As implemented, that mapping is proposed as follows:

Before|After
---|---
Common|**_Text_**
Formatting|**_Text_**
&nbsp;|Media
Layout|**_Design_**
Widgets|Widgets
Embeds|Embeds
Reusable blocks|Reusable blocks

<details><summary>Click to expand outdated notes from early iterations</summary>

Before|After
---|---
Common|Text
Formatting|Text
Layout|Design
Widgets|Tools

_(Suggested improvements to this are welcomed)_

Notably, I have not yet updated existing blocks to use an appropriate new category name, for the explicit purpose to effectively test the sensibility of this mapping arrangement, considering that this is expected to impact third-party blocks which register to one of the existing categories, at least until those plugins opt to update to a more appropriate updated category.

Prior to merge, we can consider to update the default set of core blocks to a more appropriate block category.
</details>

~As a follow-up task, I would like to explore improvements around allowing blocks to be registered without a category, or without a valid category. Such blocks should be shown in an "Uncategorized" block grouping. This could also be used to support usage where there are no default categories, and the inserter would show all blocks without any groupings. This could also be used to improve support for hypothetical future revisions to block categories, in order to better handle block registrations for plugins which support multiple older versions of WordPress.~ **Edit (2020-05-29):** This was implemented separately in #22280.

**Testing Instructions:**

Verify block categories are updated to new values, seen in the block inserter.

Verify there are no errors of block registration in your browsers Developer Tools console (i.e. all blocks should still be registered, with an appropriate category).

Optionally, verify you can change a block registration to use one of the new block category names, or a legacy category name. Legacy category usage should appear as categorized based on the mapping above.
